x-pack/legacy/plugins/ml/public/application/datavisualizer/file_based/components/import_view/importer/message_importer.js

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Importer } from './importer';
+
+export class MessageImporter extends Importer {
+  constructor(results, settings) {
+    super(settings);
+
+    this.excludeLinesRegex =
+      results.exclude_lines_pattern === undefined
+        ? null
+        : new RegExp(results.exclude_lines_pattern);
+    this.multilineStartRegex =
+      results.multiline_start_pattern === undefined
+        ? null
+        : new RegExp(results.multiline_start_pattern);
+  }
+
+  // split the text into an array of lines by looking for newlines.
+  // any lines that match the exclude_lines_pattern regex are ignored.
+  // if a newline is found, check the next line to see if it starts with the
+  // multiline_start_pattern regex
+  // if it does, it is a legitimate end of line and can be pushed into the list,
+  // if not, it must be a newline char inside a field value, so keep looking.
+  async read(text) {
```

Comment:
This is a left over from the very original csv parsing library that was used which needed to be async.
This can go, but you'll have also change the `read` method in `ndjson_importer.js` as well as removing the `await` from line 210 of `import_view.js`